api/python/cellxgene_census/src/cellxgene_census/experimental/ml/cell_dataset_builder.py

```diff
@@ -0,0 +1,135 @@
+import uuid
+from abc import ABC, abstractmethod
+from typing import Any, ContextManager, Dict, Generator, Optional, Sequence, Type
+
+import pandas as pd
+import scipy.sparse
+import tiledbsoma
+from datasets import Dataset
+
+from cellxgene_census.experimental.util import X_sparse_iter
+
+
+class CellDatasetBuilder(ContextManager["CellDatasetBuilder"], ABC):
+    """
+    Abstract base class for methods to process CELLxGENE Census query results into a
+    Hugging Face Dataset in which each item represents one cell. Subclasses implement
+    the `cell_item()` method to process the expression vector for a cell into a Dataset
+    item, and may optionally override `__init__()` and `__enter__()` to perform any
```

Comment:
```suggestion
    item. Acts as a context manager, so subclasses may optionally override `__init__()` and `__enter__()` to perform any
```
if only to explicitly mention in the docstring that it's a ContextManager

Reply:
@atolopko-czi thanks, when I got into addressing this and your other comment about obs/var terminology, I ended up significantly refactoring `CellDatasetBuilder` to inherit `ExperimentAxisQuery` directly instead of wrapping it. This simplified it significantly and I also improved the docstring showing the usage.